tiledb/sm/query/readers/sparse_index_reader_base.h

```diff
@@ -302,6 +302,14 @@ class SparseIndexReaderBase : public ReaderBase {
 
     /** Is the reader done with the query. */
     bool done_adding_result_tiles_;
+
```

Comment:
We should turn this struct to be a full class with accessors. Constructors should be documented.

Reply:
Also, we can do better than a default constructor here... frag_idx_ should be the same size as the fragment_metadata_ in the readers. Let's set that at construction.

Reply:
Done!